src/fundamental/macro-fundamental.h

```diff
@@ -327,14 +327,23 @@ static inline size_t ALIGN_TO(size_t l, size_t ali) {
         return ((l + ali - 1) & ~(ali - 1));
 }
 
+#define ALIGN2(l) ALIGN_TO(l, 2)
 #define ALIGN4(l) ALIGN_TO(l, 4)
 #define ALIGN8(l) ALIGN_TO(l, 8)
+#define ALIGN2_PTR(l) ((void*) ALIGN_TO((uintptr_t) l, 2))
+#define ALIGN4_PTR(l) ((void*) ALIGN_TO((uintptr_t) l, 4))
+#define ALIGN8_PTR(l) ((void*) ALIGN_TO((uintptr_t) l, 8))
 #ifndef SD_BOOT
 /* libefi also provides ALIGN, and we do not use them in sd-boot explicitly. */
 #define ALIGN(l)  ALIGN_TO(l, sizeof(void*))
 #define ALIGN_PTR(p) ((void*) ALIGN((uintptr_t) (p)))
 #endif
 
+/* Checks if the specified pointer is aligned as appropriate for the specific type */
+#define ALIGNED16_P(p) (((uintptr_t) p) % __alignof__(uint16_t) == 0)
+#define ALIGNED32_P(p) (((uintptr_t) p) % __alignof__(uint32_t) == 0)
+#define ALIGNED64_P(p) (((uintptr_t) p) % __alignof__(uint64_t) == 0)
```

Comment:
Why not `_PTR` for consistency?

Reply:
Also, these don't do the aligning, but the name suggests they do. Should really be called `IS_ALIGNED` or something like that.

Reply:
Oh, young grasshopper! The "_P" suffix stands for "predicate". It's a lisp thing. Don't they teach that anymore? ;-)

Basically, in lisp you suffix functions that return boolean with -p, to indicate they are a "predicate". Some of the older folks also do this in C, including glibc internally at various places. This macro is derived from the glibc internal one (which existed in the sha256.c code beforehand, see the diff), and I figured we could just stick to that.

See this:

https://jblevins.org/log/def-predicate_p
https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html
https://stackoverflow.com/questions/70545047/what-does-suffix-p-mean-in-common-lisp-functions

Some C APIs do this too, for example